### ISSUE TITLE
Add --reset option for wiping ~/.atom/storage

### ISSUE THREAD

**Issue by lee-dohm** [atom/atom#2957]
Occasionally we get instances where someone's editor has some wedged serialized state that is preventing them from opening Atom properly. When this happens, they need to be instructed on how to find `~/.atom/storage` (or the platform-specific equivalent) and to delete or move it. It would be very helpful to have an `atom --reset` command, like the `atom --safe` command that would either:

1. Delete `~/.atom/storage` prior to loading the rest of Atom
2. Package up `~/.atom/storage` and send it off to GitHub before deleting it (with the user's permission)

**Comment by maherbeg** [comment]
:+1:

In reality, if Atom crashes multiple times in a row, then that directory should be blown away. This happens often enough that I'm starting to go a bit crazy. This has been particularly apparent when working on ruby code bases for me. I never had this problem on JS code bases before.

**Comment by lee-dohm** [comment]
This will be available in Atom v1.7.x as `--clear-window-state`.

**Comment by TomCaserta** [comment]
`--safe` should automatically enable this option, had the same issue and nothing was working until I found this cli option.

**Comment by 50Wliu** [comment]
`--clear-window-state` is permanent and potentially destructive.  Associating safe mode with completely wiping out all data, including unsaved, isn't what the Atom team is going for.  Safe mode is supposed to temporarily start Atom with no community packages installed, while `--clear-window-state` is supposed to allow you to break out of situations where Atom keeps hanging/crashing because of certain editor scenarios.

**Comment by lee-dohm** [comment]
There is also complete documentation for all of this in the [Atom Flight Manual](http://flight-manual.atom.io) in the [Debugging section](http://flight-manual.atom.io/hacking-atom/sections/debugging/) including how and why to use each switch.

**Comment by TomCaserta** [comment]
Ah, thank you, that makes sense. Hadn't seen that page to be honest lee-dohm, searched google for my issue with atom crashing opening a large file but it kept coming up with people requesting option to disable the window saving feature entirely.

**Comment by tarah34** [comment]
I'm still not understanding these instructions, can you please help? I've read these and related posts via atom.io searching for resolving a bunch of tabs of files opening at startup that I haven't used for weeks.  I also read the debugging section in the Flight Manual.  However, I can't get the --clear-window-state function to work in Terminal.  Can you give easy steps (I'm a neophyte with Terminal) for how to try to resolve this problem with Atom tabbed files opening at startup that are from a very old session of using the app?  Thanks!
![screen shot 2017-01-10 at 11 19 08 am](https://cloud.githubusercontent.com/assets/23523944/21814348/07335c6e-d727-11e6-89e7-5a44e837ab89.jpg)

**Comment by lee-dohm** [comment]
@tarah34 You're doing things right from your screenshot ... for some reason the shell tools aren't installed. What you can do is execute the menu item "Atom > Install Shell Commands" inside Atom. Then completely close Atom and run the `atom --clear-window-state` from the Terminal.

**Comment by tarah34** [comment]
@lee-dohm Thanks so much!  I followed your instructions, and the dialog box message after executing "Atom > Install Shell Commands" inside Atom was a little disconcerting.  But when I tried to install shell commands a 2nd time to see whether the message would be the same, I had a different dialog box message.  I proceed to run the atom --clear-window-state from the Terminal and worked.  I've provided screenshots for the record. 👍 
![screen shot 2017-01-10 at 2 08 48 pm](https://cloud.githubusercontent.com/assets/23523944/21820855/5bd25a96-d73f-11e6-9696-3bac4709706e.jpg)

![screen shot 2017-01-10 at 2 10 20 pm](https://cloud.githubusercontent.com/assets/23523944/21820848/56853e00-d73f-11e6-9bb4-23d1163c1eed.jpg)
